src/nvim/window.c

```diff
@@ -3623,7 +3623,7 @@ win_goto_hor (
 
   if (ui_is_external(kUIWindows)) {
     // TODO(utkarshme): 0, 1 for vertical directions. 2, 3 for horizontal.
-    ui_call_win_move_cursor(2 + left, count);
+    ui_call_win_move_cursor(2 + left, count);  // calls win_goto
```

Comment:
But wouldn't that break the modularization?... We are checking for errors and return values in `api/ui.c` which feel kinda neat there.

Reply:
True that, but then you could just have it a helper function in `api/ui.c` which you call directly from here, and which just returns the window id, so `ui_win_goto` won't be needed.

Reply:
Seems perfect!

Reply:
So turns out, functions from `api/ui.c` aren't accessible from `window.c`. Moving the function to `ui.c` makes everything kind of awkward. I am more inclined to keeping the `ui_win_goto` because it makes everything more intuitive imo.

Reply:
even if you include `nvim/api/ui.h` ?  They should be.